bugbot/rules/perfalert_resolved_regression.py

```diff
@@ -0,0 +1,186 @@
+                        bug_info["resolution"] = specific_change["added"]
+                        bug_info["resolution_previous"] = (
+                            specific_change["removed"].strip() or "---"
+                        )
+                        bug_info["resolution_time"] = change["when"]
+                        break
+
+            if bug_info.get("resolution"):
+                # Find the status that the bug was resolved to, and by who
+                for specific_change in change["changes"]:
+                    if specific_change["field_name"] == "status" and specific_change[
+                        "added"
+                    ] in ("RESOLVED", "REOPENED"):
+                        bug_info["status"] = specific_change["added"]
+                        bug_info["status_author"] = change["who"]
+                        bug_info["status_time"] = change["when"]
+                        break
+
+            if bug_info.get("status"):
+                break
+
+        return bug_info
+
+    def get_resolution_info(self, bug):
+        # Match all the resolutions with resolution comments if they exist
+        bug_id = str(bug["id"])
+        bug_comments = bug["comments"]
+        bug_history = self.get_resolution_history(bug)
+
+        # Sometimes a resolution comment is not provided so use a default
+        bug_history["resolution_comment"] = DEFAULT_RESOLUTION_COMMENT
+        for comment in bug_comments[::-1]:
+            if (
+                comment["creation_time"] == bug_history["status_time"]
+                and comment["author"] == bug_history["status_author"]
+            ):
+                bug_history["resolution_comment"] = comment["text"]
+                break
+        else:
+            # Check if the bugbot has already needinfo'ed on the bug since
+            # the last status change before making one
+            skip_ni = False
+            status_time = lmdutils.get_date_ymd(bug_history["status_time"])
+            for comment in bug_comments[::-1]:
+                if lmdutils.get_date_ymd(comment["creation_time"]) <= status_time:
+                    break
+
+                if comment["author"] == BOT_MAIN_ACCOUNT:
+                    if (
+                        "comment containing a reason for why"
+                        "the performance regression"
+                        in comment["text"]
+                        or "could you provide a comment explaining the resolution?"
+                        in comment["text"]
+                    ):
+                        # Bugbot has already commented on this bug since the last
+                        # status change. No need to comment again since this was
+                        # just a resolution change
+                        skip_ni = True
+
+            if not skip_ni:
+                self.extra_ni[bug_id] = bug_history
+
+        self.extra_email_info[bug_id] = bug_history
```

Comment:
To be consistent with the rest of the project, I would move this logic here to `handle_bug()`. You can still split the logic into different methods if needed. But, we use the `data` in `handle_bug()` to save pass the data instead of `self.extra_ni` and `self.extra_email_info`.

We use `self.extra_ni` to pass extra information to the needinfo template, which is limited to only needed fields.

For `self.extra_email_info` in this case, you could pass the needed fields as columns by specifying them in the `columns()` method: https://github.com/mozilla/bugbot/blob/28320d656d6f573e990aa5583cc6d5f021dc1e0b/bugbot/rules/not_landed.py#L46-L47
Then, you can use them directly in the template in the same ordrer as you listed them in the `columns()` method: https://github.com/mozilla/bugbot/blob/28320d656d6f573e990aa5583cc6d5f021dc1e0b/templates/not_landed.html#L13
However, you should first pass these extra fields to the `data` dictionary: https://github.com/mozilla/bugbot/blob/28320d656d6f573e990aa5583cc6d5f021dc1e0b/bugbot/rules/not_landed.py#L61-L62